flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptivebatch/AdaptiveBatchSchedulerTest.java

```diff
@@ -91,6 +95,31 @@ void setUp() {
         taskRestartExecutor = new ManuallyTriggeredScheduledExecutor();
     }
 
+    @Test
+    void testExceptionHistoryWithGlobalFailureLabels() throws Exception {
+        final JobGraph jobGraph = createJobGraph();
+        final TestingFailureEnricher failureEnricher = new TestingFailureEnricher();
+        final SchedulerBase scheduler =
+                createFailingExecutionGraphScheduler(
+                        jobGraph,
+                        Collections.singleton(failureEnricher),
+                        createCustomParallelismDecider(10),
+                        BatchExecutionOptions.ADAPTIVE_AUTO_PARALLELISM_MAX_PARALLELISM
+                                .defaultValue());
+        final DefaultExecutionGraph graph = (DefaultExecutionGraph) scheduler.getExecutionGraph();
+        // Triggered failure on initializeJobVertex that should be labeled
+        try {
+            scheduler.startScheduling();
+        } catch (IllegalStateException e) {
+            // expected due to failure
+        }
```

Comment:
How about it replacing with `Assertions.assertThatThrowBy`.

Reply:
Thanks for the suggestion! Added :)